src/bsk_rl/envs/general_satellite_tasking/scenario/satellites.py

```diff
@@ -340,6 +346,16 @@ def calculate_additional_windows(self, duration: float) -> None:
         Args:
             duration: Time to calculate windows from end of previous window.
         """
+        self.logger.info(
+            "Finding opportunity windows from "
+            f"{self.window_calculation_time:.2f} to "
+            f"{self.window_calculation_time + duration:.2f} seconds"
+        )
+        if self.window_calculation_time != 0:
+            self.logger.warning(
```

Comment:
Is it possible to limit the maximum number of a given warning? I ran some cases and this one appeared several times in a row.

Reply:
Could you share an example? Didn’t see that in any of the example scripts, might be a sign of something wrong elsewhere… not good if that’s popping up a ton, inefficient and cause some inaccuracies

Reply:
Ahh, there are a lot of cases where a negative duration may be requested. The logging should come after the duration check on the next line so that it only logs in cases where windows are actually generated.